### ISSUE TITLE
Future plans

### ISSUE THREAD

**Issue by Franciman** [Franciman/vabal#24]
After an enlighting discussion on #haskell IRC channel, I think that for the future it is important to have a library that parses cabal files and projects which is _separate_ from the Cabal library, so I will try to ask cabal developers what the think about it. Another important thing to have, once this library is ready, is a mechanism to convert between different representations, for example stackage manifests, stack.yaml files hpack etc.
I think vabal could become helpful in this direction (or even disappear because it is superseded by other tools, which I'd greatly love to happen).
The second priority is, therefore, to add support for Stackage manifests to cabal, through vabal, therefore we need to understand if there is a stable representation of Stackage manifests, and learn about the current state of the art.

Finally an amusing direction I'd like to take is to transform the core vabal algorithm for choosing the ghc version to use in prolog, or at least use the logic programming paradigm to write it, I think it will be both clearer and more efficient.
